From: Eric DeVolder <eric.devolder@oracle.com>
Subject: remove ARCH_DEFAULT_KEXEC from Kconfig.kexec
Date: Wed, 2 Aug 2023 12:17:50 -0400

This patch is a minor cleanup to the series "refactor Kconfig to
consolidate KEXEC and CRASH options".

In that series, a new option ARCH_DEFAULT_KEXEC was introduced in order to
obtain the equivalent behavior of s390 original Kconfig settings for
KEXEC.  As it turns out, this new option did not fully provide the
equivalent behavior, rather a "select KEXEC" did.

As such, the ARCH_DEFAULT_KEXEC is not needed anymore, so remove it.

Link: https://lkml.kernel.org/r/20230802161750.2215-1-eric.devolder@oracle.com
Signed-off-by: Eric DeVolder <eric.devolder@oracle.com>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 arch/s390/Kconfig    |    3 ---
 kernel/Kconfig.kexec |    1 -
 2 files changed, 4 deletions(-)

--- a/arch/s390/Kconfig~remove-arch_default_kexec-from-kconfigkexec
+++ a/arch/s390/Kconfig
@@ -246,9 +246,6 @@ config PGTABLE_LEVELS
 
 source "kernel/livepatch/Kconfig"
 
-config ARCH_DEFAULT_KEXEC
-	def_bool y
-
 config ARCH_SUPPORTS_KEXEC
 	def_bool y
 
--- a/kernel/Kconfig.kexec~remove-arch_default_kexec-from-kconfigkexec
+++ a/kernel/Kconfig.kexec
@@ -17,7 +17,6 @@ config HAVE_IMA_KEXEC
 
 config KEXEC
 	bool "Enable kexec system call"
-	default ARCH_DEFAULT_KEXEC
 	depends on ARCH_SUPPORTS_KEXEC
 	select KEXEC_CORE
 	help
_
